File: candy-splitting/s.c
#include <math.h>
#include <stdio.h>

int main() {
	int cases;
	scanf("%d\n", &cases);

	int n_case;
	for (n_case = 1; n_case <= cases; n_case++) {
		int n;
		scanf("%d\n", &n);

		long long int e[n];
		int i;
		for (i = 0; i < n; i++) {
			scanf("%lld", &e[i]);
		}

		long long int t = -1;
		long long int q = 1;
		long long int il = 0;
	
		for (i = 0; i < n; i++)
			q *= 2;

		//printf("%lld\n", q);

		for (il = 1; il < q - 1; il++) {
			int total = 0;
			int wrong_total = 0;
			int wrong_total2 = 0;
			unsigned long long int mask = il;

			int j;
			for (j = 0; j < n; j++) {
				if (mask & 1) {
					//printf("%lld ", e[j]);
					total += e[j];
					wrong_total ^= e[j];
				} else {
					wrong_total2 ^= e[j];
				}
				mask = mask >> 1;
			}

			if ((wrong_total == wrong_total2) && (total > t)) {
				t = total;
			}
			//printf("\n");
		}	
		
		printf("Case #%d: ", n_case);
		if (t > 0)
			printf("%lld", t);
		else
			printf("NO");

		printf("\n");
	}
	return 0;
}

File: candy-splitting/s.py
import sys

cases = int(raw_input())

def best(list):
	m = -1
	l = len(list)
	for i in xrange(1, pow(2, l) - 1):
		total = 0
		wrong_total = 0;
		wrong_total2 = 0;

		mask = int(i)
		taken = []
		not_taken = []
		for j in xrange(l):
			if mask & 1:
				#sys.stdout.write("%d " % (list[j],))
				total += list[j]
				wrong_total ^= list[j]
			else:
				wrong_total2 ^= list[j]
			mask = mask >> 1

		#sys.stdout.write("\n")
		if (wrong_total == wrong_total2) and (total > m):
			m = total
	
	return m
		

def main():
	for case in range(1, cases + 1):
		n = int(raw_input())
		elements = [int(e) for e in raw_input().split(' ')]
		t = best(elements)
		
		sys.stdout.write("Case #%s: %s\n" % (case, t if t > 0 else 'NO'))

if __name__ == '__main__':
	main()
